pkg/api/agentPoolOnlyApi/v20180331/validate.go

```diff
@@ -130,45 +133,117 @@ func validateUniqueProfileNames(profiles []*AgentPoolProfile) error {
-			customVNETCount++
-			isCustomVNET = agentPool.IsCustomVNET()
-		}
-	}
 
-	if !(customVNETCount == 0 || customVNETCount == len(a.AgentPoolProfiles)) {
-		return fmt.Errorf("Multiple VNET Subnet configurations specified.  Each agent pool profile must all specify a custom VNET Subnet, or none at all")
+	n := a.NetworkProfile
+
+	if n == nil {
+		return ErrorNilNetworkProfile
 	}
 
-	subIDMap := make(map[string]int)
-	resourceGroupMap := make(map[string]int)
-	agentVNETMap := make(map[string]int)
-	if isCustomVNET {
-		for _, agentPool := range a.AgentPoolProfiles {
-			agentSubID, agentRG, agentVNET, _, err := GetVNETSubnetIDComponents(agentPool.VnetSubnetID)
-			if err != nil {
-				return err
-			}
+	if string(n.NetworkPlugin) == string(Azure) {
+		// validate ServiceCidr
+		_, serviceCidr, err := net.ParseCIDR(n.ServiceCidr)
+		if err != nil {
+			return ErrorInvalidServiceCidr
+		}
+
+		// validate DNSServiceIP
+		dnsServiceIP := net.ParseIP(n.DNSServiceIP)
+		if dnsServiceIP == nil {
+			return ErrorInvalidDNSServiceIP
+		}
+
+		// validate DockerBridgeCidr
+		_, _, err = net.ParseCIDR(n.DockerBridgeCidr)
+		if err != nil {
+			return ErrorInvalidDockerBridgeCidr
+		}
+
+		// validate DNSServiceIP is within ServiceCidr
+		if !serviceCidr.Contains(dnsServiceIP) {
+			return ErrorDNSServiceIPNotInServiceCidr
+		}
 
-			subIDMap[agentSubID] = subIDMap[agentSubID] + 1
-			resourceGroupMap[agentRG] = resourceGroupMap[agentRG] + 1
-			agentVNETMap[agentVNET] = agentVNETMap[agentVNET] + 1
+		// validate DNSServiceIP is not the first IP in ServiceCidr. The first IP is reserved for redirect svc.
+		if dnsServiceIP.String() == strings.TrimSuffix(serviceCidr.IP.String(), "0")+"1" {
+			return ErrorDNSServiceIPAlreadyUsed
 		}
 
-		// TODO: Add more validation to ensure all agent pools belong to the same VNET, subscription, and resource group
-		// 	if(len(subIDMap) != len(a.AgentPoolProfiles))
+		// validate custom VNET logic
+		if n.isCustomVNET() {
+			var subscription string
+			var resourceGroup string
+			var vnet string
+
+			if a.AgentPoolProfiles == nil {
+				return ErrorNilAgentPoolProfile
+			}
+
+			for _, agentPool := range a.AgentPoolProfiles {
+				if !agentPool.IsCustomVNET() {
+					return ErrorAgentPoolNoSubnet
+				}
 
-		// 	return errors.New("Multiple VNETS specified.  Each agent pool must reference the same VNET (but it is ok to reference different subnets on that VNET)")
-		// }
+				// TODO: validate VnetSubnetCidr
+
+				// TODO: validate dockerBridgeCidr is not within VnetSubnetCidr
+
+				// TODO: validate ServiceCidr does not overlap with VnetSubnetCidr
+
+				// TODO: validate VnetSubnetCidr does n ot overlap with other VnetSubnetCidr
+
+				// TODO: validate IP address pre-allocation does not exceed VnetSubnetCidr capacity
+
+				// validate subsricption, resource group and vnet are the same among subnets
+				subnetSubscription, subnetResourceGroup, subnetVnet, _, err := GetVNETSubnetIDComponents(agentPool.VnetSubnetID)
```

Comment:
Instead of doing nil checks, do a regex match of the expected resource ID (subnet).